Remove C++11 compatibility from c10::string_view

smessmer opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#30921 Remove C++11 compatibility from c10::string_view**

deletecode

Differential Revision: [D18869638](https://our.internmc.facebook.com/intern/diff/D18869638/)

smessmer committed:
Remove C++11 compatibility from c10::string_view

deletecode

Differential Revision: [D18869638](https://our.internmc.facebook.com/intern/diff/D18869638/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove C++11 compatibility from c10::string_view"


deletecode

Differential Revision: [D18869638](https://our.internmc.facebook.com/intern/diff/D18869638/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove C++11 compatibility from c10::string_view"


deletecode

Differential Revision: [D18869638](https://our.internmc.facebook.com/intern/diff/D18869638/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove C++11 compatibility from c10::string_view"


deletecode

Differential Revision: [D18869638](https://our.internmc.facebook.com/intern/diff/D18869638/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove C++11 compatibility from c10::string_view"


deletecode

Differential Revision: [D18869638](https://our.internmc.facebook.com/intern/diff/D18869638/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove C++11 compatibility from c10::string_view"


deletecode

Differential Revision: [D18869638](https://our.internmc.facebook.com/intern/diff/D18869638/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove C++11 compatibility from c10::string_view"


deletecode

Differential Revision: [D18869638](https://our.internmc.facebook.com/intern/diff/D18869638/)

[ghstack-poisoned]

kostmo commented:
## CircleCI build failures summary
As of commit accd55a:
* **13/13** failures introduced in this PR



## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=accd55a898272ef0a0e1ed7bcd47ae4d79c9db13).

### 12 new failures recognized by patterns
The following build failures don't appear to be due to upstream breakage:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3972050) pytorch_linux_xenial_py2_7_9_build (1/12)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=75587496) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=75587496))
<details>
<summary>
<code>Dec 17 01:36:35 /var/lib/jenkins/workspace/c10/util/string_view.h:195:20: error: expected unqualified-id before '::' token</code>
</summary>

```
Dec 17 01:36:35 /var/lib/jenkins/workspace/c10/util/string_view.h:113:30: error: expected primary-expression before '(' token 
Dec 17 01:36:35        throw std::out_of_range( 
Dec 17 01:36:35                               ^ 
Dec 17 01:36:35 /var/lib/jenkins/workspace/c10/util/string_view.h:115:20: error: expected unqualified-id before '::' token 
Dec 17 01:36:35          c10::guts::::to_string(pos) + ", size: " + c10::guts::::to_string(size())); 
Dec 17 01:36:35                     ^~ 
Dec 17 01:36:35 /var/lib/jenkins/workspace/c10/util/string_view.h: In member function 'constexpr c10::basic_string_view<CharT> c10::basic_string_view<CharT>::substr(c10::basic_string_view<CharT>::size_type, c10::basic_string_view<CharT>::size_type) const': 
Dec 17 01:36:35 /var/lib/jenkins/workspace/c10/util/string_view.h:193:30: error: expected primary-expression before '(' token 
Dec 17 01:36:35        throw std::out_of_range( 
Dec 17 01:36:35                               ^ 
Dec 17 01:36:35 /var/lib/jenkins/workspace/c10/util/string_view.h:195:20: error: expected unqualified-id before '::' token 
Dec 17 01:36:35          c10::guts::::to_string(pos) + ", size: " + c10::guts::::to_string(size())); 
Dec 17 01:36:35                     ^~ 
Dec 17 01:36:35 c10/CMakeFiles/c10.dir/build.make:86: recipe for target 'c10/CMakeFiles/c10.dir/core/StorageImpl.cpp.o' failed 
Dec 17 01:36:35 make[2]: *** [c10/CMakeFiles/c10.dir/core/StorageImpl.cpp.o] Error 1 
Dec 17 01:36:35 CMakeFiles/Makefile2:2012: recipe for target 'c10/CMakeFiles/c10.dir/all' failed 
Dec 17 01:36:35 make[1]: *** [c10/CMakeFiles/c10.dir/all] Error 2 
Dec 17 01:36:35 make[1]: *** Waiting for unfinished jobs.... 
Dec 17 01:36:35 [  6%] Building CXX object third_party/ideep/mkl-dnn/src/CMakeFiles/mkldnn.dir/cpu/jit_uni_dw_convolution.cpp.o 
Dec 17 01:36:35 [  6%] Building CXX object third_party/gloo/gloo/CMakeFiles/gloo.dir/common/linux.cc.o 
Dec 17 01:36:35 [  6%] Building CXX object third_party/ideep/mkl-dnn/src/CMakeFiles/mkldnn.dir/cpu/jit_sse42_convolution.cpp.o 
```

</details>

#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3972035) pytorch_macos_10_13_py3_build (2/12)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=75587698) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=75587698))
<details>
<summary>
<code>Dec 16 17:37:04 FAILED: c10/CMakeFiles/c10.dir/core/DefaultDtype.cpp.o</code>
</summary>

```
Dec 16 17:37:02 [687/2971] Building CXX object third_party/ideep/mkl-dnn/src/CMakeFiles/mkldnn.dir/cpu/rnn/ref_rnn.cpp.o 
Dec 16 17:37:02 [688/2971] Building CXX object third_party/ideep/mkl-dnn/src/CMakeFiles/mkldnn.dir/cpu/rnn/rnn_utils.cpp.o 
Dec 16 17:37:02 [689/2971] Building CXX object third_party/ideep/mkl-dnn/src/CMakeFiles/mkldnn.dir/cpu/simple_concat.cpp.o 
Dec 16 17:37:03 [690/2971] Building CXX object c10/CMakeFiles/c10.dir/core/Allocator.cpp.o 
Dec 16 17:37:03 [691/2971] Building CXX object third_party/ideep/mkl-dnn/src/CMakeFiles/mkldnn.dir/cpu/simple_sum.cpp.o 
Dec 16 17:37:03 [692/2971] Building CXX object c10/CMakeFiles/c10.dir/core/CPUAllocator.cpp.o 
Dec 16 17:37:03 [693/2971] Building CXX object c10/CMakeFiles/c10.dir/core/CopyBytes.cpp.o 
Dec 16 17:37:04 [694/2971] Linking CXX static library lib/libmkldnn.a 
Dec 16 17:37:04 [695/2971] Building CXX object c10/CMakeFiles/c10.dir/core/Device.cpp.o 
Dec 16 17:37:04 [696/2971] Building CXX object c10/CMakeFiles/c10.dir/core/DefaultDtype.cpp.o 
Dec 16 17:37:04 FAILED: c10/CMakeFiles/c10.dir/core/DefaultDtype.cpp.o  
Wno-constexpr-not-const -O3  -isysroot /Applications/Xcode-9.4.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.13.sdk -mmacosx-version-min=10.9 -fPIC   -DCAFFE2_USE_GLOO -DHAVE_GCC_GET_CPUID -DUSE_AVX -DUSE_AVX2 -DTH_HAVE_THREAD -DC10_BUILD_MAIN_LIB -fvisibility=hidden -std=gnu++14 -MD -MT c10/CMakeFiles/c10.dir/core/DefaultDtype.cpp.o -MF c10/CMakeFiles/c10.dir/core/DefaultDtype.cpp.o.d -o c10/CMakeFiles/c10.dir/core/DefaultDtype.cpp.o -c ../c10/core/DefaultDtype.cpp 
Dec 16 17:37:04 In file included from ../c10/core/DefaultDtype.cpp:1: 
Dec 16 17:37:04 In file included from ../c10/util/typeid.h:27: 
Dec 16 17:37:04 In file included from ../c10/util/TypeIndex.h:4: 
Dec 16 17:37:04 In file included from ../c10/util/ConstexprCrc.h:4: 
Dec 16 17:37:04 ../c10/util/string_view.h:115:20: error: expected unqualified-id 
Dec 16 17:37:04         c10::guts::::to_string(pos) + ", size: " + c10::guts::::to_string(size())); 
Dec 16 17:37:04                    ^ 
Dec 16 17:37:04 ../c10/util/string_view.h:195:20: error: expected unqualified-id 
Dec 16 17:37:04         c10::guts::::to_string(pos) + ", size: " + c10::guts::::to_string(size())); 
```

</details>

#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3972038) caffe2_onnx_py2_gcc5_ubuntu16_04_build (3/12)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=75587874) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=75587874))
<details>
<summary>
<code>Dec 17 01:37:12 /var/lib/jenkins/workspace/c10/util/string_view.h:195:20: error: expected unqualified-id before '::' token</code>
</summary>

```
Dec 17 01:37:12 /var/lib/jenkins/workspace/c10/util/string_view.h:113:30: error: expected primary-expression before '(' token 
Dec 17 01:37:12        throw std::out_of_range( 
Dec 17 01:37:12                               ^ 
Dec 17 01:37:12 /var/lib/jenkins/workspace/c10/util/string_view.h:115:20: error: expected unqualified-id before '::' token 
Dec 17 01:37:12          c10::guts::::to_string(pos) + ", size: " + c10::guts::::to_string(size())); 
Dec 17 01:37:12                     ^ 
Dec 17 01:37:12 /var/lib/jenkins/workspace/c10/util/string_view.h: In member function 'constexpr c10::basic_string_view<CharT> c10::basic_string_view<CharT>::substr(c10::basic_string_view<CharT>::size_type, c10::basic_string_view<CharT>::size_type) const': 
Dec 17 01:37:12 /var/lib/jenkins/workspace/c10/util/string_view.h:193:30: error: expected primary-expression before '(' token 
Dec 17 01:37:12        throw std::out_of_range( 
Dec 17 01:37:12                               ^ 
Dec 17 01:37:12 /var/lib/jenkins/workspace/c10/util/string_view.h:195:20: error: expected unqualified-id before '::' token 
Dec 17 01:37:12          c10::guts::::to_string(pos) + ", size: " + c10::guts::::to_string(size())); 
Dec 17 01:37:12                     ^ 
Dec 17 01:37:12 make[2]: *** [c10/CMakeFiles/c10.dir/core/StorageImpl.cpp.o] Error 1 
Dec 17 01:37:12 c10/CMakeFiles/c10.dir/build.make:86: recipe for target 'c10/CMakeFiles/c10.dir/core/StorageImpl.cpp.o' failed 
Dec 17 01:37:12 CMakeFiles/Makefile2:2211: recipe for target 'c10/CMakeFiles/c10.dir/all' failed 
Dec 17 01:37:12 make[1]: *** [c10/CMakeFiles/c10.dir/all] Error 2 
Dec 17 01:37:12 make[1]: *** Waiting for unfinished jobs.... 
Dec 17 01:37:12 [  9%] Generating python/operator_test/bbox_transform_test.py 
Dec 17 01:37:12 [  9%] Generating python/operator_test/clip_tensor_op_test.py 
Dec 17 01:37:12 [  9%] Building CXX object third_party/protobuf/cmake/CMakeFiles/libprotobuf.dir/__/src/google/protobuf/util/internal/field_mask_utility.cc.o 
```

</details>

#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3972047) pytorch_linux_xenial_py3_clang5_asan_build (4/12)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=75587909) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=75587909))
<details>
<summary>
<code>Dec 17 01:37:21 /var/lib/jenkins/workspace/c10/util/string_view.h:195:20: error: expected unqualified-id</code>
</summary>

```
Dec 17 01:37:21 [  6%] Building CXX object third_party/gloo/gloo/CMakeFiles/gloo.dir/rendezvous/context.cc.o 
Dec 17 01:37:21 In file included from /var/lib/jenkins/workspace/c10/core/StorageImpl.cpp:1: 
Dec 17 01:37:21 In file included from /var/lib/jenkins/workspace/c10/core/StorageImpl.h:4: 
Dec 17 01:37:21 In file included from /var/lib/jenkins/workspace/c10/core/ScalarType.h:6: 
Dec 17 01:37:21 In file included from /var/lib/jenkins/workspace/c10/util/typeid.h:27: 
Dec 17 01:37:21 In file included from /var/lib/jenkins/workspace/c10/util/TypeIndex.h:4: 
Dec 17 01:37:21 In file included from /var/lib/jenkins/workspace/c10/util/ConstexprCrc.h:4: 
Dec 17 01:37:21 /var/lib/jenkins/workspace/c10/util/string_view.h:115:20: error: expected unqualified-id 
Dec 17 01:37:21         c10::guts::::to_string(pos) + ", size: " + c10::guts::::to_string(size())); 
Dec 17 01:37:21                    ^ 
Dec 17 01:37:21 /var/lib/jenkins/workspace/c10/util/string_view.h:195:20: error: expected unqualified-id 
Dec 17 01:37:21         c10::guts::::to_string(pos) + ", size: " + c10::guts::::to_string(size())); 
Dec 17 01:37:21                    ^ 
Dec 17 01:37:21 2 errors generated. 
Dec 17 01:37:21 make[2]: *** [c10/CMakeFiles/c10.dir/core/StorageImpl.cpp.o] Error 1 
Dec 17 01:37:21 c10/CMakeFiles/c10.dir/build.make:86: recipe for target 'c10/CMakeFiles/c10.dir/core/StorageImpl.cpp.o' failed 
Dec 17 01:37:21 make[1]: *** [c10/CMakeFiles/c10.dir/all] Error 2 
Dec 17 01:37:21 make[1]: *** Waiting for unfinished jobs.... 
Dec 17 01:37:21 CMakeFiles/Makefile2:1897: recipe for target 'c10/CMakeFiles/c10.dir/all' failed 
Dec 17 01:37:21 clang: warning: argument unused during compilation: '-pie' [-Wunused-command-line-argument] 
Dec 17 01:37:21 [  6%] Building CXX object third_party/gloo/gloo/CMakeFiles/gloo.dir/rendezvous/file_store.cc.o 
```

</details>

#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3972032) pytorch_linux_xenial_py3_6_gcc5_4_build (5/12)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=75587927) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=75587927))
<details>
<summary>
<code>Dec 17 01:37:07 /var/lib/jenkins/workspace/c10/util/string_view.h:195:20: error: expected unqualified-id before '::' token</code>
</summary>

```
Dec 17 01:37:07 /var/lib/jenkins/workspace/c10/util/string_view.h:113:30: error: expected primary-expression before '(' token 
Dec 17 01:37:07        throw std::out_of_range( 
Dec 17 01:37:07                               ^ 
Dec 17 01:37:07 /var/lib/jenkins/workspace/c10/util/string_view.h:115:20: error: expected unqualified-id before '::' token 
Dec 17 01:37:07          c10::guts::::to_string(pos) + ", size: " + c10::guts::::to_string(size())); 
Dec 17 01:37:07                     ^ 
Dec 17 01:37:07 /var/lib/jenkins/workspace/c10/util/string_view.h: In member function 'constexpr c10::basic_string_view<CharT> c10::basic_string_view<CharT>::substr(c10::basic_string_view<CharT>::size_type, c10::basic_string_view<CharT>::size_type) const': 
Dec 17 01:37:07 /var/lib/jenkins/workspace/c10/util/string_view.h:193:30: error: expected primary-expression before '(' token 
Dec 17 01:37:07        throw std::out_of_range( 
Dec 17 01:37:07                               ^ 
Dec 17 01:37:07 /var/lib/jenkins/workspace/c10/util/string_view.h:195:20: error: expected unqualified-id before '::' token 
Dec 17 01:37:07          c10::guts::::to_string(pos) + ", size: " + c10::guts::::to_string(size())); 
Dec 17 01:37:07                     ^ 
Dec 17 01:37:07 c10/CMakeFiles/c10.dir/build.make:86: recipe for target 'c10/CMakeFiles/c10.dir/core/StorageImpl.cpp.o' failed 
Dec 17 01:37:07 make[2]: *** [c10/CMakeFiles/c10.dir/core/StorageImpl.cpp.o] Error 1 
Dec 17 01:37:07 make[1]: *** [c10/CMakeFiles/c10.dir/all] Error 2 
Dec 17 01:37:07 make[1]: *** Waiting for unfinished jobs.... 
Dec 17 01:37:07 CMakeFiles/Makefile2:2012: recipe for target 'c10/CMakeFiles/c10.dir/all' failed 
Dec 17 01:37:07 [  7%] Building CXX object third_party/gloo/gloo/CMakeFiles/gloo.dir/transport/tcp/unbound_buffer.cc.o 
Dec 17 01:37:07 [  7%] Building CXX object third_party/ideep/mkl-dnn/src/CMakeFiles/mkldnn.dir/cpu/ref_convolution.cpp.o 
Dec 17 01:37:07 [  7%] Building CXX object third_party/ideep/mkl-dnn/src/CMakeFiles/mkldnn.dir/cpu/jit_avx512_core_bf16_sum.cpp.o 
```

</details>

#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3972042) binary_linux_libtorch_2_7m_cpu_devtoolset7_shared-with-deps_build (6/12)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=75588022) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=75588022))
<details>
<summary>
<code>Dec 17 01:34:44 /pytorch/c10/util/string_view.h:195:20: error: expected unqualified-id before ‘::’ token</code>
</summary>

```
Dec 17 01:34:44 /pytorch/c10/util/string_view.h:113:30: error: expected primary-expression before ‘(’ token 
Dec 17 01:34:44        throw std::out_of_range( 
Dec 17 01:34:44                               ^ 
Dec 17 01:34:44 /pytorch/c10/util/string_view.h:115:20: error: expected unqualified-id before ‘::’ token 
Dec 17 01:34:44          c10::guts::::to_string(pos) + ", size: " + c10::guts::::to_string(size())); 
Dec 17 01:34:44                     ^~ 
Dec 17 01:34:44 /pytorch/c10/util/string_view.h: In member function ‘constexpr c10::basic_string_view<CharT> c10::basic_string_view<CharT>::substr(c10::basic_string_view<CharT>::size_type, c10::basic_string_view<CharT>::size_type) const’: 
Dec 17 01:34:44 /pytorch/c10/util/string_view.h:193:30: error: expected primary-expression before ‘(’ token 
Dec 17 01:34:44        throw std::out_of_range( 
Dec 17 01:34:44                               ^ 
Dec 17 01:34:44 /pytorch/c10/util/string_view.h:195:20: error: expected unqualified-id before ‘::’ token 
Dec 17 01:34:44          c10::guts::::to_string(pos) + ", size: " + c10::guts::::to_string(size())); 
Dec 17 01:34:44                     ^~ 
Dec 17 01:34:44 [  6%] Building CXX object third_party/gloo/gloo/CMakeFiles/gloo.dir/common/linux.cc.o 
Dec 17 01:34:44 [  6%] Building CXX object third_party/protobuf/cmake/CMakeFiles/libprotobuf.dir/__/src/google/protobuf/message_lite.cc.o 
Dec 17 01:34:44 [  6%] Building CXX object third_party/fbgemm/asmjit/CMakeFiles/asmjit.dir/src/asmjit/core/target.cpp.o 
Dec 17 01:34:44 gmake[2]: *** [c10/CMakeFiles/c10.dir/core/DefaultDtype.cpp.o] Error 1 
Dec 17 01:34:44 gmake[2]: *** Waiting for unfinished jobs.... 
Dec 17 01:34:44 [  6%] Building CXX object third_party/fbgemm/asmjit/CMakeFiles/asmjit.dir/src/asmjit/core/type.cpp.o 
Dec 17 01:34:44 [  6%] Building CXX object third_party/fbgemm/asmjit/CMakeFiles/asmjit.dir/src/asmjit/core/virtmem.cpp.o 
Dec 17 01:34:44 [  6%] Building CXX object third_party/gloo/gloo/CMakeFiles/gloo.dir/rendezvous/context.cc.o 
```

</details>

#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3972049) caffe2_onnx_py3_6_clang7_ubuntu16_04_build (7/12)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=75588291) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=75588291))
<details>
<summary>
<code>Dec 17 01:37:37 /var/lib/jenkins/workspace/c10/util/string_view.h:195:20: error: expected unqualified-id</code>
</summary>

```
Dec 17 01:37:37 [ 10%] Generating python/operator_test/gather_ops_test.py 
Dec 17 01:37:37 In file included from /var/lib/jenkins/workspace/c10/core/StorageImpl.cpp:1: 
Dec 17 01:37:37 In file included from /var/lib/jenkins/workspace/c10/core/StorageImpl.h:4: 
Dec 17 01:37:37 In file included from /var/lib/jenkins/workspace/c10/core/ScalarType.h:6: 
Dec 17 01:37:37 In file included from /var/lib/jenkins/workspace/c10/util/typeid.h:27: 
Dec 17 01:37:37 In file included from /var/lib/jenkins/workspace/c10/util/TypeIndex.h:4: 
Dec 17 01:37:37 In file included from /var/lib/jenkins/workspace/c10/util/ConstexprCrc.h:4: 
Dec 17 01:37:37 /var/lib/jenkins/workspace/c10/util/string_view.h:115:20: error: expected unqualified-id 
Dec 17 01:37:37         c10::guts::::to_string(pos) + ", size: " + c10::guts::::to_string(size())); 
Dec 17 01:37:37                    ^ 
Dec 17 01:37:37 /var/lib/jenkins/workspace/c10/util/string_view.h:195:20: error: expected unqualified-id 
Dec 17 01:37:37         c10::guts::::to_string(pos) + ", size: " + c10::guts::::to_string(size())); 
Dec 17 01:37:37                    ^ 
Dec 17 01:37:37 2 errors generated. 
Dec 17 01:37:37 c10/CMakeFiles/c10.dir/build.make:86: recipe for target 'c10/CMakeFiles/c10.dir/core/StorageImpl.cpp.o' failed 
Dec 17 01:37:37 make[2]: *** [c10/CMakeFiles/c10.dir/core/StorageImpl.cpp.o] Error 1 
Dec 17 01:37:37 CMakeFiles/Makefile2:2211: recipe for target 'c10/CMakeFiles/c10.dir/all' failed 
Dec 17 01:37:37 make[1]: *** [c10/CMakeFiles/c10.dir/all] Error 2 
Dec 17 01:37:37 make[1]: *** Waiting for unfinished jobs.... 
Dec 17 01:37:37 [ 10%] Generating python/operator_test/affine_channel_op_test.py 
Dec 17 01:37:37 [ 11%] Building CXX object third_party/protobuf/cmake/CMakeFiles/libprotobuf.dir/__/src/google/protobuf/util/internal/type_info.cc.o 
```

</details>

#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3972041) pytorch_linux_xenial_py3_clang5_mobile_build (8/12)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=75588716) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=75588716))
<details>
<summary>
<code>Dec 17 01:37:48 /var/lib/jenkins/workspace/c10/util/string_view.h:195:20: error: expected unqualified-id before '::' token</code>
</summary>

```
Dec 17 01:37:48 /var/lib/jenkins/workspace/c10/util/string_view.h:113:30: error: expected primary-expression before '(' token 
Dec 17 01:37:48        throw std::out_of_range( 
Dec 17 01:37:48                               ^ 
Dec 17 01:37:48 /var/lib/jenkins/workspace/c10/util/string_view.h:115:20: error: expected unqualified-id before '::' token 
Dec 17 01:37:48          c10::guts::::to_string(pos) + ", size: " + c10::guts::::to_string(size())); 
Dec 17 01:37:48                     ^ 
Dec 17 01:37:48 /var/lib/jenkins/workspace/c10/util/string_view.h: In member function 'constexpr c10::basic_string_view<CharT> c10::basic_string_view<CharT>::substr(c10::basic_string_view<CharT>::size_type, c10::basic_string_view<CharT>::size_type) const': 
Dec 17 01:37:48 /var/lib/jenkins/workspace/c10/util/string_view.h:193:30: error: expected primary-expression before '(' token 
Dec 17 01:37:48        throw std::out_of_range( 
Dec 17 01:37:48                               ^ 
Dec 17 01:37:48 /var/lib/jenkins/workspace/c10/util/string_view.h:195:20: error: expected unqualified-id before '::' token 
Dec 17 01:37:48          c10::guts::::to_string(pos) + ", size: " + c10::guts::::to_string(size())); 
Dec 17 01:37:48                     ^ 
Dec 17 01:37:48 c10/CMakeFiles/c10.dir/build.make:86: recipe for target 'c10/CMakeFiles/c10.dir/core/StorageImpl.cpp.o' failed 
Dec 17 01:37:48 make[2]: *** [c10/CMakeFiles/c10.dir/core/StorageImpl.cpp.o] Error 1 
Dec 17 01:37:48 make[1]: *** [c10/CMakeFiles/c10.dir/all] Error 2 
Dec 17 01:37:48 CMakeFiles/Makefile2:750: recipe for target 'c10/CMakeFiles/c10.dir/all' failed 
Dec 17 01:37:48 make[1]: *** Waiting for unfinished jobs.... 
Dec 17 01:37:48 [ 10%] Building CXX object third_party/gloo/gloo/CMakeFiles/gloo.dir/transport/device.cc.o 
Dec 17 01:37:48 [ 10%] Building CXX object third_party/gloo/gloo/CMakeFiles/gloo.dir/transport/pair.cc.o 
Dec 17 01:37:49 [ 10%] Building CXX object third_party/gloo/gloo/CMakeFiles/gloo.dir/transport/unbound_buffer.cc.o 
```

</details>

#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3972053) binary_linux_manywheel_2_7mu_cpu_devtoolset7_build (9/12)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=75588884) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=75588884))
<details>
<summary>
<code>Dec 17 01:34:36 /pytorch/c10/util/string_view.h:195:20: error: expected unqualified-id before ‘::’ token</code>
</summary>

```
Dec 17 01:34:36 /pytorch/c10/util/string_view.h:113:30: error: expected primary-expression before ‘(’ token 
Dec 17 01:34:36        throw std::out_of_range( 
Dec 17 01:34:36                               ^ 
Dec 17 01:34:36 /pytorch/c10/util/string_view.h:115:20: error: expected unqualified-id before ‘::’ token 
Dec 17 01:34:36          c10::guts::::to_string(pos) + ", size: " + c10::guts::::to_string(size())); 
Dec 17 01:34:36                     ^~ 
Dec 17 01:34:36 /pytorch/c10/util/string_view.h: In member function ‘constexpr c10::basic_string_view<CharT> c10::basic_string_view<CharT>::substr(c10::basic_string_view<CharT>::size_type, c10::basic_string_view<CharT>::size_type) const’: 
Dec 17 01:34:36 /pytorch/c10/util/string_view.h:193:30: error: expected primary-expression before ‘(’ token 
Dec 17 01:34:36        throw std::out_of_range( 
Dec 17 01:34:36                               ^ 
Dec 17 01:34:36 /pytorch/c10/util/string_view.h:195:20: error: expected unqualified-id before ‘::’ token 
Dec 17 01:34:36          c10::guts::::to_string(pos) + ", size: " + c10::guts::::to_string(size())); 
Dec 17 01:34:36                     ^~ 
Dec 17 01:34:36 [  1%] Building CXX object third_party/fbgemm/asmjit/CMakeFiles/asmjit.dir/src/asmjit/core/inst.cpp.o 
Dec 17 01:34:36 gmake[2]: *** [c10/CMakeFiles/c10.dir/core/DefaultDtype.cpp.o] Error 1 
Dec 17 01:34:36 gmake[1]: *** [c10/CMakeFiles/c10.dir/all] Error 2 
Dec 17 01:34:36 gmake[1]: *** Waiting for unfinished jobs.... 
Dec 17 01:34:36 [  1%] Building CXX object third_party/fbgemm/asmjit/CMakeFiles/asmjit.dir/src/asmjit/core/jitallocator.cpp.o 
Dec 17 01:34:36 [  1%] Building CXX object third_party/fbgemm/asmjit/CMakeFiles/asmjit.dir/src/asmjit/core/jitruntime.cpp.o 
Dec 17 01:34:36 [  1%] Building CXX object third_party/gloo/gloo/CMakeFiles/gloo.dir/broadcast.cc.o 
Dec 17 01:34:36 [  1%] Building CXX object third_party/fbgemm/CMakeFiles/fbgemm_avx2.dir/src/FbgemmI8Depthwise3x3Avx2.cc.o 
```

</details>

#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3972048) binary_linux_libtorch_2_7m_cpu_gcc5_4_cxx11-abi_shared-with-deps_build (10/12)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=75589642) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=75589642))
<details>
<summary>
<code>Dec 17 01:37:38 FAILED: c10/CMakeFiles/c10.dir/core/Storage.cpp.o</code>
</summary>

```
Dec 17 01:37:37        throw std::out_of_range( 
Dec 17 01:37:37                               ^ 
Dec 17 01:37:37 ../c10/util/string_view.h:195:20: error: expected unqualified-id before ‘::’ token 
Dec 17 01:37:37          c10::guts::::to_string(pos) + ", size: " + c10::guts::::to_string(size())); 
Dec 17 01:37:37                     ^ 
Dec 17 01:37:38 [697/2998] Building CXX object c10/CMakeFiles/c10.dir/core/CopyBytes.cpp.o 
Dec 17 01:37:38 [698/2998] Building CXX object c10/CMakeFiles/c10.dir/core/CPUAllocator.cpp.o 
Dec 17 01:37:38 [699/2998] Building CXX object third_party/ideep/mkl-dnn/src/CMakeFiles/mkldnn.dir/cpu/rnn/ref_postgemm_lstm.cpp.o 
Dec 17 01:37:38 [700/2998] Building CXX object third_party/ideep/mkl-dnn/src/CMakeFiles/mkldnn.dir/cpu/rnn/ref_postgemm_rnn.cpp.o 
Dec 17 01:37:38 [701/2998] Building CXX object c10/CMakeFiles/c10.dir/core/Storage.cpp.o 
Dec 17 01:37:38 FAILED: c10/CMakeFiles/c10.dir/core/Storage.cpp.o  
o-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -O3  -fPIC   -DCAFFE2_USE_GLOO -DHAVE_GCC_GET_CPUID -DUSE_AVX -DUSE_AVX2 -DTH_HAVE_THREAD -DC10_BUILD_MAIN_LIB -fvisibility=hidden -std=gnu++14 -MD -MT c10/CMakeFiles/c10.dir/core/Storage.cpp.o -MF c10/CMakeFiles/c10.dir/core/Storage.cpp.o.d -o c10/CMakeFiles/c10.dir/core/Storage.cpp.o -c ../c10/core/Storage.cpp 
Dec 17 01:37:38 In file included from ../c10/util/ConstexprCrc.h:4:0, 
Dec 17 01:37:38                  from ../c10/util/TypeIndex.h:4, 
Dec 17 01:37:38                  from ../c10/util/typeid.h:27, 
Dec 17 01:37:38                  from ../c10/core/ScalarType.h:6, 
Dec 17 01:37:38                  from ../c10/core/StorageImpl.h:4, 
Dec 17 01:37:38                  from ../c10/core/Storage.h:3, 
Dec 17 01:37:38                  from ../c10/core/Storage.cpp:1: 
Dec 17 01:37:38 ../c10/util/string_view.h: In member function ‘constexpr const CharT& c10::basic_string_view<CharT>::at(c10::basic_string_view<CharT>::size_type) const’: 
Dec 17 01:37:38 ../c10/util/string_view.h:113:30: error: expected primary-expression before ‘(’ token 
```

</details>

#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3972039) pytorch_xla_linux_xenial_py3_6_clang7_build (11/12)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=75589779) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=75589779))
<details>
<summary>
<code>Dec 17 01:38:54 /var/lib/jenkins/workspace/c10/util/string_view.h:195:20: error: expected unqualified-id</code>
</summary>

```
Dec 17 01:38:54 [  9%] Generating python/ideep/reshape_op_test.py 
Dec 17 01:38:54 In file included from /var/lib/jenkins/workspace/c10/core/StorageImpl.cpp:1: 
Dec 17 01:38:54 In file included from /var/lib/jenkins/workspace/c10/core/StorageImpl.h:4: 
Dec 17 01:38:54 In file included from /var/lib/jenkins/workspace/c10/core/ScalarType.h:6: 
Dec 17 01:38:54 In file included from /var/lib/jenkins/workspace/c10/util/typeid.h:27: 
Dec 17 01:38:54 In file included from /var/lib/jenkins/workspace/c10/util/TypeIndex.h:4: 
Dec 17 01:38:54 In file included from /var/lib/jenkins/workspace/c10/util/ConstexprCrc.h:4: 
Dec 17 01:38:54 /var/lib/jenkins/workspace/c10/util/string_view.h:115:20: error: expected unqualified-id 
Dec 17 01:38:54         c10::guts::::to_string(pos) + ", size: " + c10::guts::::to_string(size())); 
Dec 17 01:38:54                    ^ 
Dec 17 01:38:54 /var/lib/jenkins/workspace/c10/util/string_view.h:195:20: error: expected unqualified-id 
Dec 17 01:38:54         c10::guts::::to_string(pos) + ", size: " + c10::guts::::to_string(size())); 
Dec 17 01:38:54                    ^ 
Dec 17 01:38:54 2 errors generated. 
Dec 17 01:38:54 make[2]: *** [c10/CMakeFiles/c10.dir/core/StorageImpl.cpp.o] Error 1 
Dec 17 01:38:54 c10/CMakeFiles/c10.dir/build.make:86: recipe for target 'c10/CMakeFiles/c10.dir/core/StorageImpl.cpp.o' failed 
Dec 17 01:38:54 CMakeFiles/Makefile2:2012: recipe for target 'c10/CMakeFiles/c10.dir/all' failed 
Dec 17 01:38:54 make[1]: *** [c10/CMakeFiles/c10.dir/all] Error 2 
Dec 17 01:38:54 make[1]: *** Waiting for unfinished jobs.... 
Dec 17 01:38:54 [  9%] Generating python/ideep/expanddims_squeeze_op_test.py 
Dec 17 01:38:54 [  9%] Generating python/ideep/conv_transpose_test.py 
```

</details>

#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/3972051) pytorch_libtorch_linux_xenial_cuda9_cudnn7_py3_build (12/12)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=75590790) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=75590790))
<details>
<summary>
<code>Dec 17 01:39:50 ERROR:sccache::server: Compilation failed: Output { status: ExitStatus(ExitStatus(256)), stdout: "", stderr: "In file included from /var/lib/jenkins/workspace/c10/../c10/util/ConstexprCrc.h:4:0,\n                 from /var/lib/jenkins/workspace/c10/../c10/util/TypeIndex.h:4,\n                 from /var/lib/jenkins/workspace/c10/../c10/util/typeid.h:27,\n                 from /var/lib/jenkins/workspace/c10/util/typeid.cpp:1:\n/var/lib/jenkins/workspace/c10/../c10/util/string_view.h: In member function \'constexpr const CharT& c10::basic_string_view<CharT>::at(c10::basic_string_view<CharT>::size_type) const\':\n/var/lib/jenkins/workspace/c10/../c10/util/string_view.h:113:30: error: expected primary-expression before \'(\' token\n       throw std::out_of_range(\n                              ^\n/var/lib/jenkins/workspace/c10/../c10/util/string_view.h:115:20: error: expected unqualified-id before \'::\' token\n         c10::guts::::to_string(pos) + \", size: \" + c10::guts::::to_string(size()));\n                    ^\n/var/lib/jenkins/workspace/c10/../c10/util/string_view.h: In member function \'constexpr c10::basic_string_view<CharT> c10::basic_string_view<CharT>::substr(c10::basic_string_view<CharT>::size_type, c10::basic_string_view<CharT>::size_type) const\':\n/var/lib/jenkins/workspace/c10/../c10/util/string_view.h:193:30: error: expected primary-expression before \'(\' token\n       throw std::out_of_range(\n                              ^\n/var/lib/jenkins/workspace/c10/../c10/util/string_view.h:195:20: error: expected unqualified-id before \'::\' token\n         c10::guts::::to_string(pos) + \", size: \" + c10::guts::::to_string(size()));\n                    ^\n" }</code>
</summary>

```
tring_view<CharT>::size_type, c10::basic_string_view<CharT>::size_type) const\':\n/var/lib/jenkins/workspace/c10/../c10/util/string_view.h:193:30: error: expected primary-expression before \'(\' token\n       throw std::out_of_range(\n                              ^\n/var/lib/jenkins/workspace/c10/../c10/util/string_view.h:195:20: error: expected unqualified-id before \'::\' token\n         c10::guts::::to_string(pos) + \", size: \" + c10::guts::::to_string(size()));\n                    ^\n" } 
Dec 17 01:39:50  
tring_view<CharT>::size_type, c10::basic_string_view<CharT>::size_type) const\':\n/var/lib/jenkins/workspace/c10/../c10/util/string_view.h:193:30: error: expected primary-expression before \'(\' token\n       throw std::out_of_range(\n                              ^\n/var/lib/jenkins/workspace/c10/../c10/util/string_view.h:195:20: error: expected unqualified-id before \'::\' token\n         c10::guts::::to_string(pos) + \", size: \" + c10::guts::::to_string(size()));\n                    ^\n" } 
Dec 17 01:39:50  
tring_view<CharT>::size_type, c10::basic_string_view<CharT>::size_type) const\':\n/var/lib/jenkins/workspace/c10/../c10/util/string_view.h:193:30: error: expected primary-expression before \'(\' token\n       throw std::out_of_range(\n                              ^\n/var/lib/jenkins/workspace/c10/../c10/util/string_view.h:195:20: error: expected unqualified-id before \'::\' token\n         c10::guts::::to_string(pos) + \", size: \" + c10::guts::::to_string(size()));\n                    ^\n" } 
Dec 17 01:39:50  
tring_view<CharT>::size_type, c10::basic_string_view<CharT>::size_type) const\':\n/var/lib/jenkins/workspace/c10/../c10/util/string_view.h:193:30: error: expected primary-expression before \'(\' token\n       throw std::out_of_range(\n                              ^\n/var/lib/jenkins/workspace/c10/../c10/util/string_view.h:195:20: error: expected unqualified-id before \'::\' token\n         c10::guts::::to_string(pos) + \", size: \" + c10::guts::::to_string(size()));\n                    ^\n" } 
Dec 17 01:39:50  
tring_view<CharT>::size_type, c10::basic_string_view<CharT>::size_type) const\':\n/var/lib/jenkins/workspace/c10/../c10/util/string_view.h:193:30: error: expected primary-expression before \'(\' token\n       throw std::out_of_range(\n                              ^\n/var/lib/jenkins/workspace/c10/../c10/util/string_view.h:195:20: error: expected unqualified-id before \'::\' token\n         c10::guts::::to_string(pos) + \", size: \" + c10::guts::::to_string(size()));\n                    ^\n" } 
Dec 17 01:39:50  
tring_view<CharT>::size_type, c10::basic_string_view<CharT>::size_type) const\':\n/var/lib/jenkins/workspace/c10/../c10/util/string_view.h:193:30: error: expected primary-expression before \'(\' token\n       throw std::out_of_range(\n                              ^\n/var/lib/jenkins/workspace/c10/../c10/util/string_view.h:195:20: error: expected unqualified-id before \'::\' token\n         c10::guts::::to_string(pos) + \", size: \" + c10::guts::::to_string(size()));\n                    ^\n" } 
Dec 17 01:39:50  
Dec 17 01:39:50 =========== If your build fails, please take a look at the log above for possible reasons =========== 
Dec 17 01:39:50 Compile requests                51 
Dec 17 01:39:50 Compile requests executed       44 
Dec 17 01:39:50 Cache hits                      36 
Dec 17 01:39:50 Cache misses                     0 
Dec 17 01:39:50 Cache timeouts                   0 
Dec 17 01:39:50 Cache read errors                0 
Dec 17 01:39:50 Forced recaches                  0 
Dec 17 01:39:50 Cache write errors               0 
```

</details>



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

smessmer committed:
Update on "Remove C++11 compatibility from c10::string_view"


deletecode

Differential Revision: [D18869638](https://our.internmc.facebook.com/intern/diff/D18869638/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove C++11 compatibility from c10::string_view"


deletecode

Differential Revision: [D18869638](https://our.internmc.facebook.com/intern/diff/D18869638/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove C++11 compatibility from c10::string_view"


deletecode

Differential Revision: [D18869638](https://our.internmc.facebook.com/intern/diff/D18869638/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove C++11 compatibility from c10::string_view"


deletecode

Differential Revision: [D18869638](https://our.internmc.facebook.com/intern/diff/D18869638/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove C++11 compatibility from c10::string_view"


deletecode

Differential Revision: [D18869638](https://our.internmc.facebook.com/intern/diff/D18869638/)

[ghstack-poisoned]

smessmer committed:
Update on "Remove C++11 compatibility from c10::string_view"


deletecode

Differential Revision: [D18869638](https://our.internmc.facebook.com/intern/diff/D18869638/)

[ghstack-poisoned]
